plugins/woocommerce/tests/e2e-pw/tests/merchant/create-simple-product.spec.js

```diff
@@ -133,22 +133,23 @@ test.describe.serial( 'Add New Simple Product Page', () => {
 			height: '30',
 		};
 
-		await page.locator( '.shipping_options' ).click();
-		await page.locator( '#_weight' ).fill( productDimensions.weight );
-		await page.locator( '#product_length' ).fill( productDimensions.length );
-		await page.locator( '#product_width' ).fill( productDimensions.width );
-		await page.locator( '#product_height' ).fill( productDimensions.height );
+		await page.getByRole( 'link' , { name: 'Shipping' } ).click();
+		await page.getByPlaceholder( '0' ).fill( productDimensions.weight );
+		await page.getByPlaceholder( 'Length' ).fill( productDimensions.length );
+		await page.getByPlaceholder( 'Width' ).fill( productDimensions.width );
+		await page.getByPlaceholder( 'Height' ).fill( productDimensions.height );
 
 		await page
 			.frameLocator( '#excerpt_ifr' )
 			.locator( '.wp-editor' )
 			.fill( productDescriptionShort );
 
-		await page.locator( '#product_cat-add-toggle' ).click();
+		await page.getByText( '+ Add new category' ).click();
 		await page.locator( '#newproduct_cat' ).fill( Date.now().toString() );
-		await page.locator( '#product_cat-add-submit' ).click();
+		await page.getByRole( 'button', { name: 'Add new category'} ).click();
 
 		await page.locator( '#new-tag-product_tag' ).fill( productTag );
+		// await page.getByRole( 'button', { name: 'Add'} ).click();
```

Comment:
@lanej0, I will remove this commented line, but I wanted to share the findings I experienced -  this would resolve on four different elements on the page. I can go through the list of elements, and click on the right one, but `await page.locator( '.tagadd' ).click();` seems to suit better, more simple, and more robust.

Reply:
@Stojdza You can use `await page.getByRole( 'button', { name: 'Add', exact: true} ).click();` for a unique text.

Reply:
:til: `exact: true`

Thanks @veljkho!

Reply:
Yup, and it is just a general rule/guideline.  `page.locator()` isn't deprecated, it's just what we try to go with if available.  If `page.locator()` makes WAY more sense, then go that route.